packages/compiler/server/serverlib.ts

```diff
@@ -606,6 +613,31 @@ export function createServer(host: ServerHost): Server {
     }
   }
 
+  async function getDocument(params: DefinitionParams): Promise<string> {
+    const sym = await compile(params.textDocument, (program, document, file) => {
```

Comment:
Here you should really look into getting info from the `Type` not a `Symbol`. There a a `type` property in the sym.

Reply:
Careful, Sym.type is only populated for late-bound symbols, I think. addIdentifierCompletion does this:

![image](https://user-images.githubusercontent.com/75470/184715178-a2d91f99-7a8a-462e-9f74-f5f6e288636a.png)

getTypeForSymbol might be a good helper to have for this.